code/tests/functional/backend_api/conftest.py

```diff
@@ -19,8 +20,20 @@ def app_url(app_port: int) -> int:
 
 
 @pytest.fixture(scope="module")
-def app_config() -> AppConfig:
-    return AppConfig()
+def mock_httpserver(make_httpserver):
+    """
+    This is required as the default pytest httpserver fixture is scoped at the function level
+    """
+    print("Starting HTTP Mock Server")
+    server = make_httpserver
+    yield server
+    print("Stopping HTTP Mock Server")
+    server.clear()
```

Comment:
Could you please explain what this setup method is doing? I'm a bit confused by the start, then stop all within the same method.

Reply:
Heya! So this is how pytest recommeneds you create setup/teardown functions for it's tests https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended.

It runs everything before the `yield` as setup, and everything after as teardown. Adding server after `yield server` tells it what to return as part of the fixture (if anything).